### PR TITLE
Conditionally integrate some variables

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/world/map.h [hunk 1]
```diff
@@ -316,6 +316,7 @@ enum {
 extern const rct_xy16 TileDirectionDelta[];
 extern const money32 TerrainPricing[];
 
+#ifndef NO_RCT2
```
Comment (marijnvdwerf) [inline src/world/map.h]: Do these need to be conditional?

File: src/localisation/date.h [hunk 2]
```diff
@@ -49,8 +49,13 @@ extern const sint16 days_in_month[MONTH_COUNT];
 extern const rct_string_id DateFormatStringIds[];
 extern const rct_string_id DateFormatStringFormatIds[];
 
+#ifndef NO_RCT2
```
Comment (marijnvdwerf) [inline src/localisation/date.h]: DO these need to be conditional?

File: src/interface/viewport.h [hunk 3]
```diff
@@ -101,9 +96,19 @@ extern rct_viewport *g_music_tracking_viewport;
...
+extern uint8 gSavedViewZoom;
+extern uint8 gSavedViewRotation;
+extern uint8 gCurrentRotation;
 #else
 	#define unk_EE7884 RCT2_GLOBAL(0x00EE7884, paint_struct*)
 	#define unk_EE7888 RCT2_GLOBAL(0x00EE7888, paint_struct*)
+	#define gSavedViewX				RCT2_GLOBAL(RCT2_ADDRESS_SAVED_VIEW_X, sint16)
+	#define gSavedViewY				RCT2_GLOBAL(RCT2_ADDRESS_SAVED_VIEW_Y, sint16)
+	#define gSavedViewZoom			RCT2_GLOBAL(RCT2_ADDRESS_SAVED_VIEW_ZOOM, uint8)
+	#define gSavedViewRotation		RCT2_GLOBAL(RCT2_ADDRESS_SAVED_VIEW_ROTATION, uint8)
```
Comment (marijnvdwerf) [inline src/interface/viewport.h]: Do the`gSaved*` ones need to be conditional?

Comment (janisozaur) [inline src/interface/viewport.h]: Yes

File: src/localisation/date.h (hunk 2 above)
Comment (janisozaur) [inline src/localisation/date.h]: no

File: src/world/map.h (hunk 1 above)
Comment (janisozaur) [inline src/world/map.h]: no

**Comment by janisozaur** [comment]
I have updated this PR

File: src/drawing/drawing.h [hunk 4]
```diff
@@ -118,8 +118,13 @@ typedef struct rct_palette {
 
 #define PALETTE_TO_G1_OFFSET_COUNT 144
 
+#ifndef NO_RCT2
 #define gCurrentFontSpriteBase		RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_FONT_SPRITE_BASE, sint16)
 #define gCurrentFontFlags			RCT2_GLOBAL(RCT2_ADDRESS_CURRENT_FONT_FLAGS, uint16)
+#else
+extern sint16 gCurrentFontSpriteBase;
```
Comment (marijnvdwerf) [inline src/drawing/drawing.h]: What part of the code still uses this?

File: src/interface/viewport.h (hunk 3 above)
Comment (IntelOrca) [inline src/interface/viewport.h]: Why?

Comment (janisozaur) [inline src/interface/viewport.h]: If I integrate it unconditionally, only our rendering works

File: src/drawing/drawing.h (hunk 4 above)
Comment (janisozaur) [inline src/drawing/drawing.h]: I wouldn't know if there's still any code using it.

**Pull request merged**